x-pack/plugins/spaces/public/share_saved_objects_to_space/components/share_to_space_flyout_internal.tsx

```diff
@@ -204,6 +214,20 @@ export const ShareToSpaceFlyoutInternal = (props: ShareToSpaceFlyoutProps) => {
             },
             []
           ),
+          prohibitedSpaces: shareableReferences.objects.reduce((acc, x) => {
```

Comment:
Yeah as I was implementing this I was thinking that it would be better to enforce this on the client _and_ server side. The same goes for the existing validation for legacy URL aliases. 

I mentioned in my "Other notes" above that this approach isn't bullet-proof, but I'd have to make some more significant changes to validate for these on the server side too, and I was trying to get this client-side validation in place before the 8.2 release.

I opened a follow-up issue for server-side validation: #130562

> The ML plugins seems to call `updateObjectsSpaces` and exposes it's own routes, so it won't be protected. Do we foresee other plugins following a similar route?

They do, but ML objects also aren't copyable/importable, so they aren't affected.

I don't foresee other plugins following a similar route, at least not in the near term, ML was a bit of a special case for what they wanted to achieve.